src/contracts/GPv2Settlement.sol

```diff
@@ -207,12 +210,23 @@ contract GPv2Settlement {
         // amount_x * price_x = amount_y * price_y
         // ```
         // Intuitively, if a chocolate bar is 0,50€ and a beer is 4€, 1 beer
-        // is roughly worth 8 chocolate bars (`1 * 4 = 8 * 0.5`). This means
-        // that for computing the amount of token `y` equivalent to some amount
-        // of token `x` given the clearing prices, the equation is:
+        // is roughly worth 8 chocolate bars (`1 * 4 = 8 * 0.5`). From this
+        // equation, we can derive:
+        // - The limit price for selling `x` and buying `y` is respected iff
+        // ```
+        // limit_x * price_x >= limit_y * price_y
```

Comment:
I'll leave as is for now (as this is a broader discussion and outside the scope of this PR IMO). But my 2c is that this is the standard definition of prices and we should probably use it. I think what really helped me is to think of the analogy with beer and chocolate:
If I want to buy at least 7 chocolates (y) for my beer (x), and the clearing price is 4€ for beer and 0,50€ for chocolate, then:
```
1 * 4 >= 7 * 0.5
```

And my limit price is respected (as, at that clearing price, I will get 8 chocolates and I am 1 whole chocolate happier).

Reply:
Keeping prices this way also has the advantage that if we have an ETH-USDC settlement for example, we would see prices:
```
ETH: 600
USDC: 1
```

Which intuitively makes a lot of sense.

Reply:
Agreed that this is not something that should happen in this PR.

The analogy is a bit weird because `price_*` is not a price. A price has a unit of `currency1/currency2`, while here the unit is `1/currency`. So in your example (buy at least 7 chocolates (y) for my beer (x)) it's perfectly fine if we say that the beer costs 800.000$ and the chocolate 100.000$.
What we would see in practice as the result of a settlement (for a price of `1ETH = 600USD`) is in general random numbers to make the orders work:
```
ETH: 340830000
USDC: 568050
```
and it so happens that `340830000/568050 = 600`.
What I'm proposing is to get rid of this weird entity with unit `1/currency` and rephrase anything in terms of relative value. So instead of `price_chocolate = 0.5`, `price_beer = 4`, we would use a vector representing "things of equal value", so `amount_chocolate = 8`, `amount_beer = 1`, and the two amounts of thing have the same value. In the world of real numbers, it's also the case that `amount_thing = 1 / price_thing`.
I like this because it's the same thing that we are doing for `limit_x`, `limit_y`, so we don't need an extra concept of "relative" prices like it's needed currently.

This proposal is an arbitrary design decision, but I think this comment would have been easier to understand if I hadn't had to understand what a `price_*` is first.

Reply:
> The analogy is a bit weird because price_* is not a price

I disagree, I think it is a price with `price_x` in `arbitrary_quote_currency / x` and `price_y` in `arbitrary_quote_currency / y` (with my analogy being `price_beer = € / beer` and `price_chocolate = € / chocolate`.

> so amount_chocolate = 8, amount_beer = 1, and the two amounts of thing have the same value.

This would also work, then we would have a `clearingAmounts` vector.

Reply:
Then you could just use my proposed changes and still use the naming `price_x` instead of `amount_x`. In fact, `amount_x` can be seen as having a price in `x / arbitrary_base_currency`. The fact that you cannot immediately tell apart these two cases is part of why I don't like the current `price_*`.

Reply:
`x / arbitrary_base_currency` is not "price of x" though, its "price of arbitrary_base_currency".

Reply:
I opened an issue https://github.com/gnosis/gp-v2-contracts/issues/209 so that we can continue the discussion (since its not for this PR anyway).